src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -840,6 +848,24 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return [];
 	}
 
+	// Check for and disconnect from any new connections opened while in the notebook
+	// Note: notebooks should always connect with the connection URI in the following format,
+	// so that connections can be tracked accordingly throughout ADS:
+	// let connectionUri = Utils.generateUri(connection, 'notebook');
+	private async disconnectNotebookConnection(conn: ConnectionProfile): Promise<void> {
+		if (this.notebookOptions.connectionService.getConnectionUri(conn).includes(uriPrefixes.notebook)) {
```

Comment:
What if getConnectionUri returns undefined? is that possible?

Reply:
Not possible from what I could see:

return this._connectionStatusManager.getOriginalOwnerUri(Utils.generateUri(connectionProfile));